clientjs/XExt.js

```diff
@@ -1916,9 +1916,12 @@ exports = module.exports = function(jsh){
       if (options.specialKeys.escape && (e.keyCode == 13)) { e.preventDefault(); e.stopImmediatePropagation(); acceptfunc(); }
     });
     if(options.backgroundClose){
-      jsh.$root('.xdialogblock').on('click.close' + sel, function(e){
-        if(e.target != this) return;
-        if(jsh.xDialog.length && (jsh.xDialog[0]==sel)){ e.preventDefault(); e.stopImmediatePropagation(); cancelfunc(); }
+        jsh.$root('.xdialogblock').off('mousedown.close' + sel).on('mousedown.close' + sel, function(e1){
```

Comment:
Timeouts are not needed.
Mousedown and mouseup must both originate on overlay. Any clicks (either in browser or outside) will reset sequence.
This is identical to click event handling with ensuring click originates and terminates on target element.